### PR TITLE
[1.10.0 Release] Update release logs for 1.10.0

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Adds the results of the release tests for the `1.10.0` release.  

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
The previous PR (https://github.com/ray-project/ray/pull/21434) had the result of release tests run on the `1.10.0rc0` branch.  The present PR has the results from the `1.10.0` branch.  The new commits between `1.10.0rc0` and `1.10.0` shouldn't affect these results.  

However, `stress_test_dead_actors.json` and `stress_test_many_tasks.json` only had the smoke test results in the previous PR.  In this PR we include the non-smoke-test results.  It might be worth paying closer attention to these two results in this PR.

As before it might be easier to look at this diff https://github.com/ray-project/ray/pull/21512/commits/dee2c70587c3db3c60dae6730eac33504f97577a to check for regressions.

**Comment by architkulkarni** [comment]
> Do I need to review the result again?

It's up to you.  The release process doesn't have an explicit requirement for reviewing the release logs for the final wheel after they've already been reviewed for the release candidate.

However, `stress_test_dead_actors.json` and `stress_test_many_tasks.json` should probably be reviewed, since this is the first time we're seeing the results from the full test (the prior PR with `rc0` mistakenly only had the results from the smoke test.)

**Comment by architkulkarni** [comment]
Java and test_serialization flaky on master and unrelated to this PR

File: release/release_logs/1.10.0/stress_tests/stress_test_dead_actors.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "success": 1,
```
Comment (rkooo567) [inline release/release_logs/1.10.0/stress_tests/stress_test_dead_actors.json:2]: LGTM

File: release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  "stage_1_time": 216.38800859451294,
+  "stage_1_avg_iteration_time": 21.638786935806273,
+  "stage_1_max_iteration_time": 22.329787492752075,
+  "stage_1_min_iteration_time": 21.30394411087036,
+  "stage_2_time": 251.29053378105164,
+  "stage_2_avg_iteration_time": 50.257874202728274,
+  "stage_2_max_iteration_time": 51.48736071586609,
+  "stage_2_min_iteration_time": 47.86618733406067,
+  "stage_3_creation_time": 0.06580400466918945,
+  "stage_3_time": 2932.907886981964,
```
Comment (rkooo567) [inline release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:13]: Hmm this has 50% regression (`  "stage_3_creation_time": 0.05767488479614258,
  "stage_3_time": 1949.1530861854553,` -> this is 1.9). cc @ericl can you sign off if this is okay? 

Looks like the actor creation performance has regressed. cc @iycheng do you know any known issue? 

```
# Create a bunch of actors.
def stage3(total_num_remote_cpus, smoke=False):
    start_time = time.time()
    logger.info("Creating %s actors.", total_num_remote_cpus)
    actors = [Actor.remote() for _ in range(total_num_remote_cpus)]
    stage_3_creation_time = time.time() - start_time
    logger.info("Finished stage 3 actor creation in %s seconds.",
                stage_3_creation_time)

    num_tasks = 1000

    if smoke:
        num_tasks //= 25

    # Submit a bunch of small tasks to each actor. (approximately 1070 seconds)
    start_time = time.time()
    logger.info("Submitting many small actor tasks.")
    for N in [num_tasks, num_tasks * 100]:
        x_ids = []
        for i in range(N):
            x_ids = [a.method.remote(0) for a in actors]
            if i % 100 == 0:
                logger.info("Submitted {}".format(i * len(actors)))
        ray.get(x_ids)
    return time.time() - start_time, stage_3_creation_time
```

Comment (rkooo567) [reply release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:13]: So, "creation_time" is not regressed, but this just means "submission time" actually. It is time it takes to submit X number of actors asynchronously (so has no meaning).

Seems like the time X number of actors are created + submitting 1000 tasks to them has regression

Comment (iycheng) [reply release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:13]: This looks like a bad regression in my mind. I think we need to at least identify the root cause.

Comment (rkooo567) [reply release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:13]: I agree with this (especially we have a critical user who cares this metrics). Maybe we should bisect? I can create a release blocker issue. We can also see if other critical tests have regressed (e.g., many node actor test)

Comment (rkooo567) [reply release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:13]: cc @ericl @scv119 let us know your thoughts.

Comment (scv119) [reply release/release_logs/1.10.0/stress_tests/stress_test_many_tasks.json:13]: agree this is a release blocker.

**Comment by architkulkarni** [comment]
The partial fix for the regression has been merged and will be cherry picked.  There will be another round of release tests and a new version of this PR.

**Pull request closed without merging**